modules/dnn/perf/perf_layer.cpp

```diff
@@ -55,7 +55,197 @@ struct Layer_Slice : public TestBaseWithParam<tuple<Backend, Target> >
+
+        Net net;
+        LayerParams lp;
+        lp.type = "Eltwise";
+        lp.name = "testLayer";
+        lp.set("operation", op);
+        int id = net.addLayerToPrev(lp.name, lp.type, lp);
+        net.connect(0, 1, id, 1);
+
+        // warmup
+        {
+            std::vector<String> inpNames(2);
+            inpNames[0] = "a";
+            inpNames[1] = "b";
+            net.setInputsNames(inpNames);
+            net.setInput(a, inpNames[0]);
+            net.setInput(b, inpNames[1]);
+
+//            net.setPreferableBackend(backendId);
+//            net.setPreferableTarget(targetId);
+            Mat out = net.forward();
+        }
+
+        TEST_CYCLE()
+        {
+            Mat res = net.forward();
+        }
+
+        SANITY_CHECK_NOTHING();
+    }
+
+    int N = 8;
+    int C = 256;
+    int H = 128;
+    int W = 100;
+};
+
+
+PERF_TEST_P_(Layer_NaryEltwise, NCHW_NCHW_add)
+{
+    test_layer({N, C, H, W}, {N, C, H, W}, "add");
+}
+
+PERF_TEST_P_(Layer_NaryEltwise, NCHW_NCHW_div)
+{
+    test_layer({N, C, H, W}, {N, C, H, W}, "div");
+}
+
+PERF_TEST_P_(Layer_NaryEltwise, NCHW_NCHW_equal)
+{
+    test_layer({N, C, H, W}, {N, C, H, W}, "equal");
+}
+
+PERF_TEST_P_(Layer_NaryEltwise, NCHW_NCHW_greater)
+{
+    test_layer({N, C, H, W}, {N, C, H, W}, "greater");
+}
+
+PERF_TEST_P_(Layer_NaryEltwise, NCHW_NCHW_less)
+{
+    test_layer({N, C, H, W}, {N, C, H, W}, "less");
+}
+
+PERF_TEST_P_(Layer_NaryEltwise, NCHW_NCHW_max)
+{
+    test_layer({N, C, H, W}, {N, C, H, W}, "max");
+}
+
+PERF_TEST_P_(Layer_NaryEltwise, NCHW_NCHW_mean)
+{
+    test_layer({N, C, H, W}, {N, C, H, W}, "mean");
+}
+
+PERF_TEST_P_(Layer_NaryEltwise, NCHW_NCHW_min)
+{
+    test_layer({N, C, H, W}, {N, C, H, W}, "min");
+}
+
+PERF_TEST_P_(Layer_NaryEltwise, NCHW_NCHW_mul)
+{
+    test_layer({N, C, H, W}, {N, C, H, W}, "mul");
+}
+
+PERF_TEST_P_(Layer_NaryEltwise, NCHW_NCHW_pow)
+{
+    test_layer({N, C, H, W}, {N, C, H, W}, "pow");
+}
+
+PERF_TEST_P_(Layer_NaryEltwise, NCHW_NCHW_sub)
+{
+    test_layer({N, C, H, W}, {N, C, H, W}, "sub");
+}
+
+PERF_TEST_P_(Layer_NaryEltwise, NCHW_NCHW_sum)
+{
+    test_layer({N, C, H, W}, {N, C, H, W}, "sum");
+}
+
+PERF_TEST_P_(Layer_NaryEltwise, NCHW_NCHW_ref_mul)
+{
+    test_ref_layer({N, C, H, W}, {N, C, H, W}, "prod");
+}
+
+PERF_TEST_P_(Layer_NaryEltwise, NCHW_NCHW_ref_sum)
+{
+    test_ref_layer({N, C, H, W}, {N, C, H, W}, "sum");
+}
+
+PERF_TEST_P_(Layer_NaryEltwise, NCHW_NCHW_ref_max)
+{
+    test_ref_layer({N, C, H, W}, {N, C, H, W}, "max");
+}
+
+PERF_TEST_P_(Layer_NaryEltwise, NCHW_NCHW_ref_min)
+{
+    test_ref_layer({N, C, H, W}, {N, C, H, W}, "min");
+}
+
+PERF_TEST_P_(Layer_NaryEltwise, NCHW_NCHW_ref_div)
+{
+    test_ref_layer({N, C, H, W}, {N, C, H, W}, "div");
+}
+
+PERF_TEST_P_(Layer_NaryEltwise, NCHW_C_sum)
+{
+    test_layer({N, C, H, W}, {C, 1, 1}, "sum");
+}
+
+PERF_TEST_P_(Layer_NaryEltwise, NHWC_C)
+{
+    test_layer({N, H, W, C}, {1, C}, "sum");
+}
```

Comment:
Might I suggest alternating pattern(non-ref/ref layers) where applicable, so that it's easier to compare the results in the output?

Reply:
Done with putting non-ref and ref layers together. 

```cpp
PERF_TEST_P_(Layer_NaryEltwise, NCHW_NCHW_div)
{
    test_layer({N, C, H, W}, {N, C, H, W}, "div");
}

PERF_TEST_P_(Layer_NaryEltwise, NCHW_NCHW_ref_div)
{
    test_layer({N, C, H, W}, {N, C, H, W}, "div", true);
}
```